app/viewmodels/helpers/ViewMovementActionsHelper.scala

```diff
@@ -0,0 +1,131 @@
+      link(appConfig.emcsTfeChangeDestinationUrl(request.ern, movement.arc), "viewMovement.changeDestination", Some("submit-a-change-of-destination"), hintKey = Some("viewMovement.changeDestination.info"))
+    }
+  }
+
+  def alertOrRejectionLink(movement: GetMovementResponse)(implicit request: DataRequest[_], messages: Messages): Option[Html] = {
+    Option.when(AlertOrRejectValidStatuses.contains(movement.eadStatus)) {
+      link(appConfig.emcsTfeAlertOrRejectionUrl(request.ern, movement.arc), "viewMovement.alertOrRejection", Some("submit-alert-or-rejection"), hintKey = Some("viewMovement.alertOrRejection.info"))
+    }
+  }
+
+  def reportOfReceiptLink(movement: GetMovementResponse)(implicit request: DataRequest[_], messages: Messages): Option[Html] = {
+    Option.when(ReportOfReceiptValidStatuses.contains(movement.eadStatus)) {
+      link(appConfig.emcsTfeReportAReceiptUrl(request.ern, movement.arc), "viewMovement.reportAReceipt", Some("submit-report-of-receipt"), hintKey = Some("viewMovement.reportAReceipt.info"))
+    }
+  }
+
+  def explainADelayLink(movement: GetMovementResponse)(implicit request: DataRequest[_], messages: Messages): Option[Html] = {
+    Option.when(cond = true) {
+      link(appConfig.emcsTfeExplainDelayUrl(request.ern, movement.arc), "viewMovement.explainDelay", Some("explain-a-delay"), hintKey = Some("viewMovement.explainDelay.info"))
+    }
+  }
+
+  def shortageOrExcessLink(movement: GetMovementResponse)(implicit request: DataRequest[_], messages: Messages): Option[Html] = {
+    val linkValid = movement.destinationType match {
+      case Export => ShortageOrExcessExportValidStatuses.contains(movement.eadStatus)
+      case _      => ShortageOrExcessValidStatuses.contains(movement.eadStatus)
+    }
+
+    Option.when(linkValid) {
+      link(appConfig.emcsTfeExplainShortageOrExcessUrl(request.ern, movement.arc), "viewMovement.explainShortageOrExcess", Some("explain-shortage-or-excess"), hintKey = Some("viewMovement.explainShortageOrExcess.info"))
+    }
+  }
+
+  def printLink()(implicit messages: Messages): Option[Html] =
+    Option.when(cond = true) {
+      link("print", "viewMovement.printOrSaveEad", Some("print-or-save-ead"), hintKey = Some("viewMovement.printOrSaveEad.info"))
+    }
+}
+
+object ViewMovementActionsHelper {
+  private val CancelMovementValidStatuses = Seq("Accepted", "Exporting", "Rejected")
+  private val ChangeDestinationValidStatuses = Seq("Accepted", "Exporting", "PartiallyRefused", "Refused", "DeemedExporting", "Rejected")
+  private val AlertOrRejectValidStatuses = Seq("Accepted")
+  private val ReportOfReceiptValidStatuses = Seq("Accepted")
+  private val ShortageOrExcessValidStatuses = Seq("Delivered", "Diverted", "ManuallyClosed", "Refused", "PartiallyRefused", "Exporting", "Stopped", "DeemedExported")
+  private val ShortageOrExcessExportValidStatuses = Seq("Delivered", "Diverted", "ManuallyClosed", "Refused", "PartiallyRefused", "Exporting", "Stopped", "Accepted", "Rejected")
```

Comment:
I don't think you need to do this as part of this PR; but it would be good at somepoint to make this more typesafe and have an ADT for these. (even if it's just an Enum.)

So that there's a proper `MovementStatus` model which has concrete implementations of the different possible statuses. Then update our `GetMovementResponse` model so that the status is parsed at the point when we retrieve the movement.

Having this as a `String` everywhere is loosely typed at the moment.

Reply:
I agree. We have all of these over in `MovementFilterStatusOption` but I choose not to use it as the name is too much related to the filtering of movements.

Reply:
https://github.com/hmrc/emcs-tfe-frontend/pull/134/commits/debc880ea7194d4ad67d4f1a69d22114589fda93